lib/most_frequent_integer.rb

```diff
@@ -3,5 +3,28 @@
 # exists, return nil.
 
 def find_most_frequent_integer(arr)
-  # put your code here
+  #use a Hash to keep track of the number of duplicates
+  new_hash = Hash.new(0)
+  #iterate through each element in the array
+  arr.each do |idx|
```

Comment:
Lines 9-12 look great. This is a small detail, but I'd like to encourage you to use descriptive variable names -- in an `each` loop, we are working with the _element_ in an Array (I misspoke on this when we were working together the other evening). `idx` is a bit misleading, as it makes me think we are dealing with the index.